src/api/verification_case.py

```diff
@@ -0,0 +1,430 @@
+                    unique_hash = uuid.uuid1()
+                    case["case_id_in_suite"] = str(unique_hash)
+                    self.case_suite[uuid.uuid1()] = case
+            else:
+                return None
+
+        if file_path is not None:
+            # check 'file_path' type
+            if self.check_type("file_path", file_path, str):
+                # check if json or directory path is provided
+                if self.is_json_path_type(file_path):
+                    # check if 'file_path' exists
+                    if self.check_file("file_path", file_path):
+                        self.case_suite = self.read_case(file_path, self.case_suite)[0]
+                    else:
+                        return None
+
+                else:  # when directory path is provided
+                    # check if the directory exists
+                    if os.path.exists(file_path):
+                        # find all the JSON files in file_path directory
+                        json_file_names = glob.glob(os.path.join(file_path, "*.json"))
+
+                        # check if JSON file(s) exists.
+                        if len(json_file_names) != 0:
+                            for file in json_file_names:
+                                self.case_suite = self.read_case(
+                                    file_path, self.case_suite
+                                )[0]
+                        else:
+                            logging.warning(
+                                "There is no JSON files in the given file_path."
+                            )
+                    else:
+                        logging.error(
+                            f"The provided directory doesn't exist. Please make sure to provide a correct file_path."
+                        )
+                        return None
+            else:
+                return None
+
+    def load_verification_cases_from_json(
+        self, json_case_path: str = None
+    ) -> List[str]:
+        """Add verification cases from specified json file into self.case_suite
+
+        Args:
+            json_case_path: str, path to the json file containing fully defined verification cases.
+
+        Returns:
+          List, unique ids of verification cases loaded in self.case_suite
+        """
+
+        # check `json_case_path` type
+        if self.check_type("json_case_path", json_case_path, str):
+            # check if `json_case_path` exists
+            if self.check_file("json_case_path", json_case_path):
+                self.case_suite, newly_added_hash = self.read_case(
+                    json_case_path, self.case_suite
+                )
+            else:
+                return None
+        else:
+            return None
+
+        return newly_added_hash
+
+    def save_case_suite_to_json(
+        self, json_path: str = None, case_ids: List = []
+    ) -> NoReturn:
```

Comment:
Which is None?

Question) In this case, is it correct to set the return type as ```UNION[NoReturn, None]```? Because there is a case there is no return?

Reply:
No Return means this method does not return. E.g. it will raise exception within. return None is not NoReturn.

https://stackoverflow.com/a/3459131

Reply:
I understand the difference between the ```None``` return and ```NoReturn```.

The gist of my question was considering the code structure, there could be a case where there is no return and a case where there is a ```None``` return because if the method is executed correctly, there is no return and if for example a wrong```json_path``` argument is provided, there is a None return.

So, is it correct to set the return type hint as ```UNION[NoReturn, None]```? Also, could you check the stackoverflow link? it seems like the link is not relevant for this discussion.

Reply:
I did not see a case with NoReturn. Unless I missed something, when the method is executed correctly, it returns None although it was not written explicitly.

Reply:
great point! Fixed.